app/models/hostgroup.rb

```diff
@@ -323,4 +324,14 @@ def validate_subnet_types
     errors.add(:subnet, _("must be of type Subnet::Ipv4.")) if subnet.present? && subnet.type != 'Subnet::Ipv4'
     errors.add(:subnet6, _("must be of type Subnet::Ipv6.")) if subnet6.present? && subnet6.type != 'Subnet::Ipv6'
   end
+
+  def validate_compute_profile
+    return if compute_profile_id.nil?
+    compute_profile = ComputeProfile.find(compute_profile_id)
+    errors.add(:compute_profile, _('is not valid.')) unless valid_compute_profiles.include? compute_profile
+  end
+
+  def valid_compute_profiles
+    ComputeProfile.authorized(:view_compute_profiles).visibles.to_a
+  end
```

Comment:
When editing an existing hg, the current user might not have permission to view the compute profile - In the UI form, we use https://github.com/theforeman/foreman/blob/develop/app/helpers/hosts_and_hostgroups_helper.rb#L20-L22 to include the current profile in the list.

Reply:
I think including the current profile in the list will create a different issue - when creating a hostgroup from hammer if you specify an non-valid compute_profile it is the current profile and if we will add it to the list the validation will not work well.
I hope that this is explained clear. I assume we can find another way to deal with this. What do you say?